src/generateIgnoreFile.js

```diff
@@ -32,6 +32,30 @@ const localSourceFetcher = async (block, directory) => {
   )
   return joinLinesWithEOF(files)
 }
+const relativeSourceFetcher = async (block, directory) => {
+  const files = await promiseMap(
+    async (relativeFilePath) => {
+      const filePath = path.dirname(relativeFilePath)
+      const fileContent = await readFile(path.join(directory, relativeFilePath))
+      const splittedFileContent = fileContent.split(LINE_BREAK)
+      const edittedFileContent = splittedFileContent.map(line => {
+        if (!line) {
+          return line
+        }
+
+        if (line.startsWith('!')) {
```

Comment:
I think there are (at least) a couple more special cases you need here, looking at https://git-scm.com/docs/gitignore#_pattern_format.

- If the line starts with a `#`, then it's a comment, so don't prefix it.

- If there is no `/` separator at the start or middle (but end doesn't count), then the pattern is directory recursive, so e.g. I think `*.js` should become `<path>/**/*.js` instead of `<path>/*.js`.

Reply:
The `#` case was cover in a later commit:

https://github.com/foray1010/ignore-sync/commit/cbbcbd0038db1cbb7bac1f331feedef4151dc373#diff-426ca2249f092e513fabf419f7f466a5b314ce8d1dbef79f2e6a4fd231510f83R45

Reply:
And for the other case it may be better to create a new issue.

Reply:
Filed https://github.com/foray1010/ignore-sync/issues/302